samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -0,0 +1,192 @@
+		mosh_error("Error requesting notification of prediction availability: %d", err);
+	}
+}
+#endif /* defined(CONFIG_NRF_CLOUD_PGPS) */
+
+static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
+{
+	int err = 0;
+
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTING:
+		mosh_print("NRF_CLOUD_EVT_TRANSPORT_CONNECTING");
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTED:
+		mosh_print("NRF_CLOUD_EVT_TRANSPORT_CONNECTED");
+		break;
+	case NRF_CLOUD_EVT_READY:
+		mosh_print("NRF_CLOUD_EVT_READY");
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED:
+		mosh_print("NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED");
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		mosh_print("NRF_CLOUD_EVT_ERROR");
+		break;
+	case NRF_CLOUD_EVT_SENSOR_DATA_ACK:
+		mosh_print("NRF_CLOUD_EVT_SENSOR_DATA_ACK");
+		break;
+	case NRF_CLOUD_EVT_FOTA_DONE:
+		mosh_print("NRF_CLOUD_EVT_FOTA_DONE");
+		break;
+	case NRF_CLOUD_EVT_RX_DATA:
+		mosh_print("NRF_CLOUD_EVT_RX_DATA");
+		if (((char *)evt->data.ptr)[0] == '{') {
+			/* Not A-GPS data */
+			break;
+		}
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+		err = nrf_cloud_agps_process((char *)evt->data.ptr, evt->data.len);
+		if (!err) {
+			mosh_print("A-GPS data processed");
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+			/* call us back when prediction is ready */
+			k_work_submit(&notify_pgps_work);
+#endif
+			/* data was valid; no need to pass to other handlers */
+			break;
+		}
+#endif
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+		err = nrf_cloud_pgps_process(evt->data.ptr, evt->data.len);
+		if (err) {
+			mosh_error("Error processing P-GPS packet: %d", err);
+		}
+#else
+		if (err) {
+			mosh_print("Unable to process A-GPS data, error: %d", err);
+		}
+#endif
+		break;
+	case NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST:
+		mosh_print("NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST");
+		mosh_warn("Add the device to nRF Cloud and wait for it to reconnect");
+		break;
+	case NRF_CLOUD_EVT_USER_ASSOCIATED:
+		mosh_print("NRF_CLOUD_EVT_USER_ASSOCIATED");
+		break;
+	default:
+		mosh_error("Unknown nRF Cloud event type: %d", evt->type);
+		break;
+	}
+}
+
+static void cloud_connect(void)
+{
+	int err;
+
+	struct nrf_cloud_init_param config = {
+		.event_handler = nrf_cloud_event_handler,
+	};
+
+	err = nrf_cloud_init(&config);
+	if (err == -EACCES) {
+		mosh_print("nRF Cloud module already initialized");
+	} else if (err) {
+		mosh_error("nrf_cloud_init, error: %d", err);
+		return;
+	}
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	else { /* Cloud module successfully initialized */
+		/* Done here, the work structure only gets initialized once */
+		k_work_init(&notify_pgps_work, notify_pgps);
+	}
+#endif
+
+	err = nrf_cloud_connect(NULL);
+	if (err == NRF_CLOUD_CONNECT_RES_ERR_ALREADY_CONNECTED) {
+		mosh_print("nRF Cloud connection already established");
+	} else if (err) {
+		mosh_error("nrf_cloud_connect, error: %d", err);
+	} else {
+		mosh_print("Connection to nRF Cloud established");
+		mosh_print("Endpoint: %s", CONFIG_NRF_CLOUD_HOST_NAME);
```

Comment:
I'd make these into one line print unless there is specific reason to have two lines?